documentation/src/main/asciidoc/ch09.asciidoc

```diff
@@ -370,6 +370,100 @@ include::{sourcedir}/org/hibernate/validator/referenceguide/chapter09/Bootstrapp
 Refer to <<section-fail-fast>> and <<section-programmatic-api>> to learn more about the fail fast
 mode and the constraint declaration API.
 
+[[section-script-evaluator-factory]]
+==== Configuring script evaluator factory (`ScriptEvaluatorFactory`)
+
+For constrains like `@ScriptAssert` and `@ParameterScriptAssert` it might be useful to configure
+how scripts are evaluated, and which scripting engines are used. This can be done by setting
+a custom implementation of `ScriptEvaluatorFactory`. In particular this is important for
+OSGi environments (or similar to it) where user might face issues with dynamic class loading
+and https://www.jcp.org/en/jsr/detail?id=223[JSR-223]. It also would allow to use any custom
+script evaluation, not necessarily based on the https://www.jcp.org/en/jsr/detail?id=223[JSR-223].
+
+===== Programmatic configuration
+There are two ways to change the default `ScriptEvaluatorFactory` programmatically.
+First one is to use `hibernate.validator.script_evaluator_factory` property as described
+in <<section-provider-specific-settings>> section, by passing a fully qualified class name of `ScriptEvaluatorFactory`
+implementation to be used. In this case configuration will look like:
+
+[[example-script-evaluator-factory-via-addproperty]]
+.Changing `ScriptEvaluatorFactory` via `addProperty()`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter09/BootstrappingTest.java[tags=scriptEvaluatorFactoryViaAddProperty]
+----
+====
+
+Another option is to pass an instance of `ScriptEvaluatorFactory` when building `ValidatorFactory`.
+This gives more flexibility in configuration of the `ScriptEvaluatorFactory`. <<example-script-evaluator-factory-programmatically>>
+shows how this can be done.
```

Comment:
Should be reworked because of the remark above. But just as I think this part will be kept: "when building **the** `ValidatorFactory`"